frontend/rust-lib/flowy-grid/src/services/field/type_options/number_type_option/number_type_option.rs

```diff
@@ -97,10 +98,14 @@ impl NumberTypeOptionPB {
 
     pub(crate) fn format_cell_data(&self, s: &str) -> FlowyResult<NumberCellData> {
         match self.format {
-            NumberFormat::Num => match Decimal::from_str(s) {
-                Ok(value, ..) => Ok(NumberCellData::from_decimal(value)),
-                Err(_) => Ok(NumberCellData::new()),
-            },
+            NumberFormat::Num => {
+                let re = Regex::new(r"[^\d\.]").unwrap();
```

Comment:
You can try lazy_static. For example:
```
lazy_static! {
    static ref URL_REGEX: Regex = Regex::new(
        "[(http(s)?):\\/\\/(www\\.)?a-zA-Z0-9@:%._\\+~#=]{2,256}\\.[a-z]{2,6}\\b([-a-zA-Z0-9@:%_\\+.~#?&//=]*)"
    )
    .unwrap();
}
```

Reply:
ah, thanks for suggestion, this is the good point, i was consider lazy_static, but then i think the Regex is not to expensive to compute, so we don't really need initialized lazily. i think so.

Reply:
That depends. 
```
#[cfg(test)]
mod tests {
    use fancy_regex::Regex;
    use std::time::Instant;

    #[test]
    fn time_cost_test() {
        let start = Instant::now();
        let re = Regex::new(r"[^\d\.]").unwrap();
        let duration = start.elapsed();
        println!("Time elapsed in expensive_function() is: {:?}", duration);
    }
}

```
The result is `3ms` test on my computer.
<img width="698" alt="image" src="https://user-images.githubusercontent.com/86001920/210723178-84073792-6f51-46ac-ad14-4e8dff1835e4.png">


So if we want to get 1000 cells' content using the `format_cell_data`, that will be expensive.

Reply:
nice, i see, let me switch to use lazy_static!

Reply:
done, @appflowy please help me review again